src/database.c

```diff
@@ -220,8 +227,8 @@ static int write_db(const char *index, const char *data)
 	MDB_val key, value;
 	MDB_txn *txn;
 	int rc;
-    size_t len;
-    char *hash;
+	size_t len;
+	char *hash = NULL;
```

Comment:
There really isn't a need to initialize hash to NULL. It's set to either NULL or some other value if `len > MAX_PATH_LEN`.

Reply:
The only reason why I did that is to avoid **-Wmaybe-uninitialized** warnings.

Reply:
Yes, and hence the reason why I was alluding to the fact that shoving this chunk of code into a helper and calling it unconditionally would be much nicer. But as you wish, you can do the refactoring later. :-)